cedar-policy/src/tests.rs

```diff
@@ -3375,6 +3375,45 @@ mod schema_based_parsing_tests {
             Err(EntitiesError::TransitiveClosureError(_))
         ));
     }
+
+    /// Record inside entity doesn't conform to schema
```

Comment:
Perhaps in addition to this test (with an extra attribute on the inner record), we should also have a test where an attribute on the inner record is missing? maybe also both of those for a schema where `foo` is optional?

Reply:
Done